sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -106,6 +111,25 @@ public DocTableInfo getWritableTable(TableIdent tableIdent) {
             throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
                 "%s is an alias. Write, Drop or Alter operations are not supported", tableInfo.ident()));
         }
+        if (docTableInfo.closed()) {
+            throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                "The table %s is closed. Write, Drop or Alter operations (other than OPEN) are not supported",
+                tableInfo.ident()));
+        }
+        return docTableInfo;
+    }
+
+    public DocTableInfo getOpenableOrClosableTable(TableIdent tableIdent) {
```

Comment:
yes pls adopt that all alter table operation will call this method instead, this will/should reflect the real permissions/blocks